legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/DependenciesCache.java

```diff
@@ -91,43 +100,85 @@ private Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv, Functio
-                    Set<ProjectVersion> deps = this.transitiveDependencies.getIfAbsentPut(dpv,() ->
+                    DependencyResult deps = this.transitiveDependencies.getIfAbsentPut(dep,() ->
                     {
                         absentKeys.getAndIncrement();
-                        return getTransitiveDependencies(dpv, projectDataProvider);
+                        return calculateTransitiveDependencies(dep, projectDataProvider, status);
                     });
-                    dependencies.addAll(deps);
+                    if (deps.getStatus() == DependencyStatus.FAIL)
+                    {
+                        throw new IllegalStateException(String.format("Dependency not present in store %s", dep.getGav()));
+                    }
+                    dependencies.add(dep);
+                    dependencies.addAll(deps.getProjectVersion());
                 });
             }
+            else
+            {
+                return new DependencyResult(DependencyStatus.FAIL, dependencies);
+            }
         }
         catch (Exception e)
         {
             LOGGER.error("error getting transitive dependencies {}",e.getMessage());
+            return new DependencyResult(DependencyStatus.FAIL, dependencies);
         }
-        return dependencies;
+        return new DependencyResult(status, dependencies);
     }
 
     public Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv)
     {
         if (pv.getVersionId().equals(MASTER_SNAPSHOT))
         {
             absentKeys.getAndIncrement();
-            Set<ProjectVersion> masterSNAPSHOTDependencies = getTransitiveDependencies(pv, getProjectDataFromStore());
-            this.transitiveDependencies.put(pv,masterSNAPSHOTDependencies);
-            return masterSNAPSHOTDependencies;
+            DependencyResult depResult = calculateTransitiveDependencies(pv, getProjectVersionDataFromStore(), DependencyStatus.SUCCESS);
+            if (depResult.status == DependencyStatus.FAIL)
+            {
+                throw new RuntimeException(String.format("Error fetching dependencies for %s", pv.getGav()));
+            }
+            this.transitiveDependencies.put(pv,depResult);
+            return depResult.getProjectVersion();
         }
         else
         {
-            //we might end up with dirty entries for wrong projects or non existent versions in the cache but we need speed
-            //we must absolutely return empty dependencies
-            return this.transitiveDependencies.getIfAbsentPut(pv, () ->
+            DependencyResult depResult = this.transitiveDependencies.getIfAbsentPut(pv, () ->
             {
                 absentKeys.getAndIncrement();
-                return getTransitiveDependencies(pv, getProjectDataFromStore());
+                return calculateTransitiveDependencies(pv, getProjectVersionDataFromStore(), DependencyStatus.SUCCESS);
             });
+            if (depResult.status == DependencyStatus.FAIL)
```

Comment:
we should retry here . in case the missing dependency issue has been resolved.. of still mssing it will throw the error